src/lib/WalletStore.ts

```diff
@@ -88,79 +54,61 @@ export class WalletStore {
-        const db = await this.connect();
-        const transaction = db.transaction([WalletStore.DB_ACCOUNTS_STORE_NAME]);
-        const request = transaction.objectStore(WalletStore.DB_ACCOUNTS_STORE_NAME).get(id);
-        const result = await WalletStore._requestAsPromise(request, transaction);
+        this._storeName = WalletStore.DB_ACCOUNTS_STORE_NAME;
+        const result = await super.get(id);
         return result ? WalletInfo.fromObject(result) : result;
     }
 
     public async put(walletInfo: WalletInfo) {
-        const db = await this.connect();
-        const transaction = db.transaction([WalletStore.DB_ACCOUNTS_STORE_NAME], 'readwrite');
-        const request = transaction.objectStore(WalletStore.DB_ACCOUNTS_STORE_NAME).put(walletInfo.toObject());
-        return WalletStore._requestAsPromise(request, transaction);
+        this._storeName = WalletStore.DB_ACCOUNTS_STORE_NAME;
+        return super.put(walletInfo.toObject());
     }
 
     public async remove(id: string) {
-        const db = await this.connect();
-        const transaction = db.transaction([WalletStore.DB_ACCOUNTS_STORE_NAME], 'readwrite');
-        const request = transaction.objectStore(WalletStore.DB_ACCOUNTS_STORE_NAME).delete(id);
-        return WalletStore._requestAsPromise(request, transaction);
+        this._storeName = WalletStore.DB_ACCOUNTS_STORE_NAME;
+        return super.remove(id);
     }
 
     public async list(): Promise<WalletInfoEntry[]> {
-        const db = await this.connect();
-        const request = db.transaction([WalletStore.DB_ACCOUNTS_STORE_NAME], 'readonly')
-            .objectStore(WalletStore.DB_ACCOUNTS_STORE_NAME)
-            .openCursor();
+        this._storeName = WalletStore.DB_ACCOUNTS_STORE_NAME;
+        return super.list()  as Promise<WalletInfoEntry[]>;
+    }
 
-        const result: WalletInfoEntry[] = await WalletStore._readAllFromCursor(request);
-        return result;
+    public getConstants(): StoreConstants {
+        return {
+            DB_VERSION: 1,
+            DB_NAME: 'nimiq-hub',
+            DB_STORE_NAME: this._storeName,
+        };
+    }
+
+    protected upgrade(request: any, event: IDBVersionChangeEvent): void {
+        const db = request.result;
+        if (event.oldVersion < 1) {
+            // Version 1 is the first version of the database.
+            db.createObjectStore(WalletStore.DB_ACCOUNTS_STORE_NAME, { keyPath: 'id' });
+            db.createObjectStore(WalletStore.DB_META_DATA_STORE_NAME, { keyPath: 'name' });
+        }
     }
 
-    public async close() {
-        if (!this._dbPromise) { return; }
-        // If failed to open database (i.e. dbPromise rejects) we don't need to close the db
-        const db = await this._dbPromise.catch(() => null);
-        this._dbPromise = null;
-        if (db) { db.close(); }
+    protected infoToEntry(walletInfoOrValue: WalletInfoEntry): WalletInfoEntry {
+        return walletInfoOrValue;
```

Comment:
This is just a hand-through function. Could we make this the native behavior of the `Store` class, so it doesn't need to be re-implemented?

Reply:
I expect this to be the exception. In general it will be more like it is done in `CashlinkStore`.

Reply:
Ok, I looked at it more thoroughly now. 

This is actually a workaround for the fact that `WalletStore` has 2 object stores in the same database. The conversion between `InfoEntry` and `Info` is done explicitly in this case (cmp: https://github.com/nimiq/hub/blob/sebastian/cashlink-store/src/lib/WalletStore.ts#L59), instead of using these 2 Functions within `Store<Info, InfoEntry>`.